omod/src/main/java/org/openmrs/module/fhirExtension/web/mapper/TaskMapper.java

```diff
@@ -100,14 +103,30 @@ public TaskResponse constructResponse(Task task) {
 		response.setComment(task.getFhirTask().getComment());
 		return response;
 	}
-	
+
 	public void fromRequest(TaskUpdateRequest taskUpdateRequest, Task task) {
 		FhirTask fhirTask = task.getFhirTask();
-		
+
 		fhirTask.setStatus(taskUpdateRequest.getStatus());
 		fhirTask.setExecutionStartTime(taskUpdateRequest.getExecutionStartTime());
 		fhirTask.setExecutionEndTime(taskUpdateRequest.getExecutionEndTime());
 		fhirTask.setComment(taskUpdateRequest.getComment());
 	}
-	
+
+	private Concept getConceptForTaskType(String taskType) {
+		if (taskType == null || taskType.isEmpty()) {
+			log.warn("Task type is not passed. Setting as null");
+			return null;
+		}
+		Concept conceptForTaskType = Context.getConceptService().getConceptByName(taskType);
```

Comment:
This conceptService method will consider userLocale first. 

If you expect the taskType to be provided as "user input" then the above is good. 

Alternatively, if it is to be specified by implementation locale, this might be a better option. `conceptService.getConceptsByName(name, locale, false)`
where you get the LocalUtility.getDefaultLocale() 

If this method is called by both human and non-human user inputs - then 
> conceptService.getConcepts(conceptName, localeList, false, null, null, null, null, null, 0, null)

Might be better.